static-instrumenter/agent-instrumenter/src/test/java/io/opentelemetry/contrib/staticinstrumenter/ClassArchiveTest.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.staticinstrumenter;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.jar.JarFile;
+import java.util.zip.ZipOutputStream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+public class ClassArchiveTest {
+
+  @TempDir private File tempDir;
+
+  @Test
+  void shouldCopyAllClasses(@TempDir File destination) throws IOException {
+
+    // given
+    byte[] newTransformed = new byte[] {(byte) 0xCA, (byte) 0xFE, (byte) 0xBA, (byte) 0xBE};
+    Map<String, byte[]> transformed = Collections.singletonMap("test/TestClass", newTransformed);
+    String jar = JarTestUtil.createJar(tempDir, "one.jar", "test/TestClass.class", "second.class");
+    ClassArchive underTest = new ClassArchive(new JarFile(jar), transformed);
+
+    // when
+    File jarOut = new File(destination, "one.jar");
```

Comment:
Let's use a different filename for the input and output jar to reduce confusion

Reply:
The idea here was that under normal conditions out jars will have the same name as input jars (just other folder). But I understand your point.